### PR TITLE
Refactor `test_terminate_on_nan` to use `pytest.mark.parametrize`

### LINKED ISSUE #1469: Parametrize metric tests
## 🚀 Feature

Use `pytest.mark.parametrize` for metric testing

### PR DESCRIPTION
**Pull request by Moh-Yakoub**
Fixes #1469 - partly

Description:

I noticed that the test `test_terminate_on_nan_and_inf` was repeating the test cases and could have utilized the `pytest` parameterize decorator.

Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format
- [x] Documentation is updated (if required)

### CODE REVIEW

**Comment by Moh-Yakoub** [comment]
@vfdev-5 any thoughts on this one?

**Comment by sdesrozis** [comment]
@Moh-Yakoub Thank you! A test failed https://github.com/pytorch/ignite/pull/1717/checks?check_run_id=2050379447

could you check why ? Thanks!

File: tests/ignite/handlers/test_terminate_on_nan.py [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
 
-def test_terminate_on_nan_and_inf():
+@pytest.mark.parametrize(
+    "state_output,should_terminate",
+    [
+        (1.0, False),
+        (torch.tensor(123.45), False),
+        (torch.asin(torch.randn(10,)), True),
+        (torch.asin(torch.randn(4, 4)), True),
+        ((10.0, 1.0 / torch.randint(0, 2, size=(4,)).type(torch.float), 1.0), True),
```
Comment (vfdev-5) [inline tests/ignite/handlers/test_terminate_on_nan.py:16]: Let's replace random data by deterministic:
```suggestion
        ((10.0, 1.0 / torch.tensor([1.0, 2.0, 0.0, 3.0]), 1.0), True),
```

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @Moh-Yakoub !
